programs/local/customvm/main.go

```diff
@@ -0,0 +1,199 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/ava-labs/avalanche-network-runner/local"
+	"github.com/ava-labs/avalanche-network-runner/network"
+	"github.com/ava-labs/avalanche-network-runner/utils"
+	"github.com/ava-labs/avalanche-network-runner/vms"
+	avagoconfig "github.com/ava-labs/avalanchego/config"
+	"github.com/ava-labs/avalanchego/utils/logging"
+	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
+)
+
+const (
+	healthyTimeout = 2 * time.Minute
+	subnetTimeout  = 2 * time.Minute
+	binaryPathKey  = "binaryPath"
+	vmPathKey      = "vmPaths"
+	genesisPathKey = "genesisPaths"
+	subnetIDKey    = "subnetIds"
+	vmIDKey        = "vmIds"
+	loglevelKey    = "logLevel"
+)
+
+var (
+	goPath            = os.ExpandEnv("$GOPATH")
+	defaultBinaryPath = fmt.Sprintf("%s%s", goPath, "/src/github.com/ava-labs/avalanchego/build/avalanchego")
+)
+
+type customVMConfig struct {
+	BinaryPath   string   `json:"binaryPath"`
+	VMPaths      []string `json:"vmPaths"`
+	GenesisPaths []string `json:"genesisPaths"`
+	SubnetIDs    []string `json:"subnetIds"`
+	VMIDs        []string `json:"vmIds"`
+}
+
+// Shows example usage of the Avalanche Network Runner with custom VMs.
+// Creates a local five node Avalanche network all starting an avalanchego binary
+// and a provided custom VM binary. Waits for all nodes to become healthy.
+// The network runs until the user provides a SIGINT or SIGTERM.
+// Example of how to run this:
+// go run programs/local/customvm/main.go --vm-path "/path/to/vm/binary" --genesis-path "/path/to/genesis/file" --subnet-ids "24tZhrm8j8GCJRE9PomW8FaeqbgGS4UAQjJnqqn8pq5NwYSYV1" --vm-ids "tGas3T58KzdjLHhBDMnH2TvrddhqTji5iZAMZ3RXs2NLpSnhH"
+func main() {
```

Comment:
nit: can `main` call a function `run()` instead of having application logic inside `main`? This makes it possible to just have one place where we do `os.Exit(1)` (namely, if `run` returns an error) instead of several places. See [here](https://github.com/ava-labs/avalanche-testing-new/blob/main/main.go#L56) for an example